regressors/include/iDynTree/Regressors/DynamicsRegressorGenerator.h

```diff
@@ -0,0 +1,312 @@
+
+    /**
+     * Get a human readable description of all the outputs.
+     *
+     */
+    std::string getDescriptionOfOutputs();
+
+    /**
+     * Get a human readable description of a given inertal degree of freedom of the robot.
+     *
+     */
+    std::string getDescriptionOfDegreeOfFreedom(int dof_index);
+
+    /**
+     * Get a human readable description of all the internal degrees of freedom of the robot.
+     *
+     * @return a std::string containing the description of the internal degrees of freedom.
+     */
+    std::string getDescriptionOfDegreesOfFreedom();
+
+    /**
+     * Get a human readable description of a given link considered in the regressor.
+     *
+     */
+    std::string getDescriptionOfLink(int link_index);
+
+    /**
+     * Get a human readable description of all links considered in the regressor.
+     *
+     * @return a std::string containing the description of the .
+     */
+    std::string getDescriptionOfLinks();
+
+    /**
+     * Get name of the base link.
+     *
+     */
+    std::string getBaseLinkName();
+
+
+    //@}
+
+
+    /**
+     * @name Methods to access the underlyng model for the robot and the sensors
+     * (todo)
+     */
+    //@{
+
+    //const UndirectedTree & getRobotModel();
+    //const SensorsTree    & getSensorsModel();
+
+    //@}
+
+    /**
+      * @name Methods to submit the input data for calculating the regressor and the known terms
+      * This methods are used to submit the input data (such as the robot state and the sensor readings)
+      * to calculate the regressor and the known terms
+      */
+    //@{
+
+    /**
+     * Set the state for the robot (floating base)
+     * \todo TODO define a RobotState class.
+     */
+    bool setRobotState(const iDynTree::VectorDynSize& q,
+                       const iDynTree::VectorDynSize& q_dot,
+                       const iDynTree::VectorDynSize& q_dotdot,
+                       const iDynTree::Transform &world_T_base,
+                       const iDynTree::Twist& base_velocity,
+                       const iDynTree::Twist& base_acceleration
+                       const iDynTree::Twist& world_gravity);
+
+    /**
+     * Set the state for the robot (fixed base)
+     * Same as setRobotState, but with:
+     *  world_T_base      = iDynTree::Transform::Identity()
+     *  base_velocity     = iDynTree::Twist::Zero();
+     *  base_acceleration = iDynTree::Twist::Zero();
+     *
+     * \todo TODO define a RobotState class.
+     */
+    bool setRobotState(const iDynTree::VectorDynSize &q,
+                       const iDynTree::VectorDynSize &q_dot,
+                       const iDynTree::VectorDynSize &q_dotdot,
+                       const iDynTree::Twist& world_gravity);
+
+    /**
+     * Get a reference to the SensorsMeasurements structure, where
+     * sensors readings can be set for use of the dynamics regressor generator.
+     */
+    //SensorsMeasurements & getSensorsMeasurements();
```

Comment:
Ah yes! I forgot to uncomment it.. basically all the sensor measure can be set in the SensorMeasurements class (so multiple classes that need to set/get sensor measurement can avoid duplicating this functionalities).